src/components/accessories/patientExams/PatientExams.tsx

```diff
@@ -0,0 +1,188 @@
+    deleteLabReset,
+    updateLabReset,
+    activityTransitionState,
+  ]);
+
+  const labStore = useSelector<IState, ILaboratoriesState>(
+    (state: IState) => state.laboratories
+  );
+  const exams = useSelector((state: IState) => state.exams.examList.data);
+  const onSubmit = (lab: LaboratoryDTO, rows: string[]) => {
+    setShouldResetForm(false);
+    lab.patientCode = patientData?.code;
+    lab.exam = exams?.find((item) => item.code === lab.exam);
+    lab.patName = patientData?.firstName + " " + patientData?.secondName;
+    lab.sex = patientData?.sex;
+    lab.age = patientData?.age;
+    lab.examDate = lab.date;
+    lab.inOutPatient = "R";
+    if (labToEdit.code) lab.code = labToEdit.code;
+    const labWithRowsDTO = {
+      laboratoryDTO: lab,
+      laboratoryRowList: rows,
+    };
+    if (!creationMode && labToEdit.code) {
+      dispatch(updateLab(labToEdit.code, labWithRowsDTO));
+    } else {
+      dispatch(createLab(labWithRowsDTO));
+    }
+  };
+
+  const onEdit = (row: LaboratoryDTO) => {
+    dispatch(getExamRows(row.exam?.code ?? ""));
+    setLabToEdit(row);
+    setCreationMode(false);
+    scrollToElement(null);
+  };
+
+  const onDelete = (code: number | undefined) => {
+    setDeletedObjCode(`${code}` ?? "");
+    dispatch(deleteLab(code));
+  };
+
+  const resetFormCallback = () => {
+    setShouldResetForm(false);
+    setShouldUpdateTable(false);
+    setCreationMode(true);
+    dispatch(createLabReset());
+    dispatch(updateLabReset());
+    dispatch(deleteLabReset());
+    setActivityTransitionState("IDLE");
+    scrollToElement(null);
+  };
+
+  return (
+    <div className="patientExam">
+      <ExamForm
+        fields={
+          creationMode
+            ? initialFields
+            : updateLabFields(initialFields, labToEdit)
+        }
+        onSubmit={onSubmit}
+        submitButtonLabel={creationMode ? t("common.save") : t("common.update")}
+        resetButtonLabel={t("common.discard")}
+        shouldResetForm={shouldResetForm}
+        resetFormCallback={resetFormCallback}
+        isLoading={
+          labStore.createLab.status === "LOADING" ||
+          labStore.updateLab.status === "LOADING"
+        }
+      />
+      {labStore.deleteLab.status === "LOADING" && (
+        <CircularProgress style={{ marginLeft: "50%", position: "relative" }} />
+      )}
+
+      {(labStore.createLab.status === "FAIL" ||
+        labStore.updateLab.status === "FAIL" ||
+        labStore.deleteLab.status === "FAIL") && (
+        <div ref={infoBoxRef} className="info-box-container">
+          <InfoBox type="error" message={t("common.somethingwrong")} />
+        </div>
+      )}
+
+      <PatientExamsTable
+        handleEdit={onEdit}
+        handleDelete={onDelete}
+        shouldUpdateTable={shouldUpdateTable}
+      />
+
+      <ConfirmationDialog
+        isOpen={
+          labStore.createLab.status === "SUCCESS" ||
+          labStore.updateLab.status === "SUCCESS"
+        }
```

Comment:
See the https://github.com/informatici/openhospital-ui/pull/166#discussion_r685064910